internal/agent/instrumentation/instrumentation.go

```diff
@@ -29,14 +81,29 @@ func (p *pprofServer) Run(ctx context.Context) {
 	}
 }
 
-// NewPprofServer wires a core PprofServer if profiling is enabled.
-func NewPprofServer(l *log.PrefixLogger, cfg *agent_config.Config) *pprofServer {
-	ps := &pprofServer{log: l}
+// AgentInstrumentation is a thin facade that runs metrics + pprof in parallel.
+type AgentInstrumentation struct {
+	metrics *metricsServer
+	pprof   *pprofServer
+	log     *log.PrefixLogger
+}
 
-	if cfg == nil || !cfg.ProfilingEnabled {
-		return ps
+// NewAgentInstrumentation builds the agent’s observability instrumentation (e.g., metrics, pprof).
+func NewAgentInstrumentation(l *log.PrefixLogger, cfg *agent_config.Config) *AgentInstrumentation {
+	return &AgentInstrumentation{
+		metrics: NewMetricsServer(l, cfg),
+		pprof:   NewPprofServer(l, cfg),
+		log:     l,
 	}
+}
 
-	ps.srv = coreinst.NewPprofServer(l)
-	return ps
+// Run starts all observability components (e.g., metrics, pprof) and blocks until ctx is canceled.
+func (ai *AgentInstrumentation) Run(ctx context.Context) {
+	var wg sync.WaitGroup
+	wg.Add(2)
+
+	go func() { defer wg.Done(); ai.metrics.Run(ctx) }()
+	go func() { defer wg.Done(); ai.pprof.Run(ctx) }()
```

Comment:
what happens if one of these goroutines returns do we know? should we shutdown the other?